docs/v3/core/index.rst

```diff
@@ -1599,46 +1471,266 @@ Storage transformers may be stacked to combine different functionalities:
+objects with just a `name` key.
+
+.. _extension-definition-must-understand:
+
+`must_understand`
+^^^^^^^^^^^^^^^^^
+
+An extension object is interpreted to have an implicit field `must_understand` set to
+`True`, unless otherwise stated. An extension object MAY explicitly set `must_understand=False` if
+implementations can ignore its presence.
+
+An implementation MUST fail to open Zarr groups or arrays if any
+metadata fields are present which (a) the
+implementation does not recognize and (b) are not explicitly
+set to ``"must_understand": false``.
+
+`must_understand=False` is not supported for the following extension points:
+data type, chunk grid, and chunk key encoding.
+
+Use of `must_understand=False` to add top-level keys is discouraged in favor
+of the explicit use of :ref:`extension-points`.
+
+.. _extension-naming:
+
+Extension naming
+----------------
+
+The `name` field of an extension is an identifier taking one of two forms:
+**registered names** (as simple strings) and **unregistered names** (as URIs).
+
+Implementations SHOULD be able to resolve multiple names to the same
+implementation to support unregistered names which are subsequently registered.
+
+.. _extension-naming-registered-names:
+
+Registered names
+^^^^^^^^^^^^^^^^
+
+Registered names consist of a single string that is unique within the Zarr ecosystem, with no prefix.
+Registered names are intended for well-known extensions aimed at broad adoption and maximum interoperability.
+Registered names are unique and immutable.
+Registered names MUST start with one lower case letter a-z and then be followed
+by only lower case letters a-z, numerals 0-9, underscores, dots and dashes.
+
+Registered names MUST be assigned within a central repository, `zarr-extensions`_
+a Github repository, where extensions and their specification are listed.
+The Zarr Steering Council or by delegation a
+maintainer team reserves the right to refuse name assignment at its own
+discretion.
+
+- **Accepted regex:** ``^[a-z][a-z0-9-_.]+$``
+- **Valid examples:**
+  - ``zstd``
+  - ``numcodecs.adler32``
+- **Invalid examples:**
+  - ``foo/bar``
+
+.. _extension-naming-unregistered-names:
+
+Unregistered names
+^^^^^^^^^^^^^^^^^^
+
+Unregistered names are intended for private extensions and for experimental and development purposes.
+
+Unregistered names are not centrally managed and MAY be used by any extension without coordination.
+
+Unregistered names consist of URIs, which are prefixed with a scheme beginning
+with a letter and followed by any number of letters, numbers, plus symbols,
+dashes or dots and then followed by a colon.
+
+- **Identifying regex:** ``^([a-z][a-z0-9-_]+\.)+[a-z][a-z0-9-_]+:``
+
+URIs (`Uniform Resource Identifiers <https://en.wikipedia.org/wiki/Uniform_Resource_Identifier>`_)
+are a well-known mechanism to identify resources on the internet and extension authors are
+encouraged to explore further documentation on which identifiers might best express their intent.
```

Comment:
The language about resources on the internet here is confusing. I can create a zarr extension locally, on my own computer, without connecting to the internet. That suggests that zarr extensions are not really resources on the internet. Rather, zarr extensions are _code_, which might have a specification / source code on the internet, but those documents are not the same thing as the extension itself.

Reply:
Stealing from the wikipedia article:
```suggestion
URIs (`Uniform Resource Identifiers <https://en.wikipedia.org/wiki/Uniform_Resource_Identifier>`_)
are a well-known mechanism to identify abstract or phyiscal resources and extension authors are
encouraged to explore further documentation on which identifiers might best express their intent.
```